libbpf-tools/ext4dist.c

```diff
@@ -153,8 +153,8 @@ static bool should_fallback(void)
 	 * to fall back to use kprobe to be compatible with the old kernel.
 	 */
 	if (is_kernel_module("ext4") && !access("/sys/kernel/btf/ext4", R_OK))
-		return true;
-	return false;
+		return false;
+	return true;
```

Comment:
as I mentioned on another PR, I think this is wrong. We fallback to kprobe if ext4 is a module and kernel doesn't support module BTFs. That's what current logic is doing, and I think correctly.

Reply:
>as I mentioned on another PR, I think this is wrong. We fallback to kprobe if ext4 is a module and kernel doesn't support module BTFs. That's what current logic is doing, and I think correctly.

I misunderstood the objective of this function. Thanks.

BTW,
If kernel doesn't support module BTFs, I think `access("/sys/kernel/btf/ext4", R_OK)` will not return 0 but return -1.
So, I think this check is ` if (is_kernel_module("ext4") && (access("/sys/kernel/btf/ext4", R_OK) == -1 ))`

Reply:
you are right, `!access()` is wrong here. It should be just `is_kernel_module() && access()` (no need for == -1, enough to check for non-zero return).

Reply:
Thank for very much for your support.
I fixed it.